feat: Implementing the possibility to remove a friend

yassine04e opened the pull request:
1) Adding a list of friends in the modifiying profile screen, with clickable cards to delete a friend 
2) Adding the function in the repository. It uses transactions

RemIsMyWaifuu commented on app/src/main/java/com/github/se/travelpouch/ui/profile/ProfileScreen.kt:
```diff
@@ -103,54 +102,55 @@ fun ProfileScreen(navigationActions: NavigationActions, profileModelView: Profil
               modifier = Modifier.fillMaxSize().padding(horizontal = 8.dp),
-              contentPadding = PaddingValues(bottom = 80.dp)) {
-                val friends = profile.value.friends.keys.toList()
+              // contentPadding = PaddingValues(bottom = 80.dp)
```
Make sure not to leave any commented out lines like this

yassine04e replied:
This is done

RemIsMyWaifuu commented on app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepositoryFirebase.kt:
```diff
@@ -217,6 +217,46 @@ class ProfileRepositoryFirebase(private val db: FirebaseFirestore) : ProfileRepo
+            val friendProfileUpdated =
+                friendProfile.copy(friends = friendProfile.friends - userProfile.email)
+
+            db.runTransaction { t ->
+                  t.update(it.reference, "friends", friendProfileUpdated.friends)
+                  t.update(documentReference!!, "friends", userProfileUpdated.friends)
+                }
+                .addOnSuccessListener {
+                  Log.d("DeleteFriend", "Friend deleted")
+                  onSuccess(userProfileUpdated)
+                }
+                .addOnFailureListener {
+                  Log.e("DeleteFriend", "An error occurred updating user profile")
+                  onFailure(Exception("An error occurred updating your profile"))
+                }
```
Good use of transactions, and the error handling is succinct and logical.

RemIsMyWaifuu commented on app/src/main/java/com/github/se/travelpouch/ui/profile/ModifyingProfileScreen.kt:
```diff
@@ -140,74 +148,163 @@ fun ModifyingProfileScreen(
+              if (openDialogToDeleteFriend) {
+                Dialog(onDismissRequest = { openDialogToDeleteFriend = false }) {
+                  Box(
+                      Modifier.fillMaxWidth(1f)
+                          .height(250.dp)
```
While it's the same size as for when adding a friend, the dialog box is too big for the content inside.
Either increase the size of the content or reduce the size of the dialog box. It feels weird.
![Screenshot_2024-12-05_20-52-01](https://github.com/user-attachments/assets/cacae485-98b0-4a14-9c8a-cda9c7e67eb1)

yassine04e replied:
I have corrected it

RemIsMyWaifuu commented on app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepositoryFirebase.kt:
```diff
@@ -217,6 +217,46 @@ class ProfileRepositoryFirebase(private val db: FirebaseFirestore) : ProfileRepo
         .addOnFailureListener { onFailure(Exception("getting friend profile failed")) }
   }
 
+  override fun removeFriend(
```
Please add some minimal documentation for this function

yassine04e replied:
Done

RemIsMyWaifuu commented on app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepository.kt:
```diff
@@ -55,4 +55,11 @@ interface ProfileRepository {
       onFailure: (Exception) -> Unit
   )
+
+  fun removeFriend(
```
Please add minimal documentation to this function

yassine04e replied:
Done

RemIsMyWaifuu commented on app/src/main/java/com/github/se/travelpouch/model/profile/ProfileModelView.kt:
```diff
@@ -85,4 +85,15 @@ class ProfileModelView @Inject constructor(private val repository: ProfileReposi
         })
   }
+
+  fun removeFriend(friendUid: String, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
```
Please add minimal documentation to this function

yassine04e replied:
Done

RemIsMyWaifuu reviewed (changes requested):
Apart from some UI issues that need to be resolved, the backend and tests are sublime.

RemIsMyWaifuu commented:
Furthermore, the text in ProfileScreen is not aligned the same way as in ModifyingProfileScreen. This causes the "Friends :" text to be aligned on the left, but then aligned in the middle once a user enters editing mode. This is strange.

`Column(
        verticalArrangement = Arrangement.spacedBy(8.dp),
        modifier =
            Modifier.fillMaxWidth()
                .padding(horizontal = 16.dp, vertical = 16.dp)
                .padding(pd)
                .testTag("ProfileColumn"))`
and

`Column(
            verticalArrangement = Arrangement.spacedBy(8.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            modifier =
                Modifier.fillMaxWidth()
                    .padding(horizontal = 16.dp, vertical = 16.dp)
                    .padding(pd)
                    .testTag("ProfileColumn")
                    .verticalScroll(rememberScrollState()))`

RemIsMyWaifuu commented:
Also, the add friend button is on the bottom right. On the edit travel screen the way we decided to address this was to make a drawer button. Since this is not necessary here, please change the colour to be the same blend as in EditTravelSettings and [Optional] perhaps put the button under the friends list instead.

yassine04e commented:
Concerning the two other comments, I have applied the modifications you requested. I changed the floating button to add a friend, to instead put a button similar to the one to save our profile. It is placed just above the save button. 
Also, I have corrected the horizontal alignement of the column in the profile screen. Now it is the same as the edit profile screen.

RemIsMyWaifuu commented:
While technically the scope of this PR concerns itself only with friend removal, when a user saves their profile all their friends disappear..
All the other requested changes were addressed, I also removed some commented lines.
Approved.
Closes #238
